tests/init/crash/i12128/Reflect_1.scala

```diff
@@ -0,0 +1,30 @@
+trait Context: // Dummy scala.reflect.macros.Context
```

Comment:
This regression test using the dummy `scala.reflect.macros.Context` for some reason does not trigger the issue reported in #12128.  It passes when compiled with 3.0.0 (aside from the cyclic reference while unpickling if -Ycheck is specified).

So, I have manually verified that the issue is fixed, but the test passes even without the fix in place.  :confused: 

It's the end of my day here, but I can probably find time to investigate this tomorrow if needed.

Reply:
> This regression test using the dummy scala.reflect.macros.Context for some reason does not trigger the issue reported in #12128.
> So, I have manually verified that the issue is fixed

Thanks for checking that. I intended to add the original example as a test, but as it depends on `scala-reflect`, which makes it difficult to do so --- creating a dummy `Context` seems to be the closest thing that can be done.

Reply:
I have some (mostly complete) changes that I believe will make the test case work properly without the dependency on `scala-reflect`. I'll finish and push them in the morning.

Reply:
Considering the regression test consists of three components:

  (a) Reflect_1 (as a stand-in for `scala-reflect`)
  (b) Macro_2 (as a stand-in for `munit`)   
  (c) Test_3 (user code)

the key to reproduce the issue in #12128 is to compile these separately, with

  - the output from (a) on the classpath while compiling (b)
  - the output from (b) on the classpath while compiling (c)
  - the output from (a) *_not_* on the classpath while compiling (c)

In the real-world case of #12128, this situation occurs because MUnit has a `Provided` dependency on `scala-reflect`:
https://github.com/scalameta/munit/blob/d592d71f8360f1eaae2f19c1d582b75fe5d44714/build.sbt#L178-L182

I have rebased (to resolve conflicts) and pushed a commit which modifies the testing to meet the above requirements.

Incidentally, #12128 seems to have been fixed both by #12484 (for the old init checker) and #12495 (the new init checker).